Runtimes: support installing the swiftdoc for the modules

compnerd opened the pull request:
Add support to install the swiftdoc into the installed image. This is required to allow us to package up the runtime on Windows.

compnerd committed:
Runtimes: refactor swift interface installation

Extract the `install_swift_interface` function into a new file. This
is always invoked to perform the installation of the Swift interfaces as
well as the binary module. Additionally, it prepares for a future
refactoring which will also install the swift documentation interface.

compnerd commented:
@swift-ci please smoke test

etcwilde reviewed (commented):
Do we want to make it a separate function or should we roll it into part of the module installation? Are there libraries where we don't want to include the swiftdoc when we do include the module?

compnerd commented:
> Do we want to make it a separate function or should we roll it into part of the module installation? Are there libraries where we don't want to include the swiftdoc when we do include the module?

Good question. I don't know the best answer here. I think that we do always want to install the swiftdoc. However, I don't think that this should be part of the `emit_swift_interface` call. I think that we could rename that to `install_swift_module` and merge the two though.

I can see the value in keeping it separate - `install(TARGETS ...)` will need to have a flag to install the swift doc I imagine. I can also see the value in merging it - it is nice to have a single install module call that takes care of all the details.

compnerd commented:
@swift-ci please test Windows platform

compnerd commented:
@swift-ci please smoke test

compnerd committed:
Runtimes: install the `.swiftdoc` interface as well

This is part of the development component for the module and is part of
the interface definition. This content is consumed by the user during
development.

compnerd commented:
@swift-ci please smoke test
